backend/models/AssignmentGroup.js

```diff
@@ -0,0 +1,33 @@
+import mongoose from "mongoose";
+
+const assignmentGroupSchema = new mongoose.Schema({
```

Comment:
Is this better than having "tags" on an assignment?

Reply:
Yeah since there's other metadata like grading weight and grading policy that's related to the assignment type. I definitely would have used the tags method otherwise. This hierarchy also makes sense for the way we're gonna fetch / compute data at the frontend.